operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java

```diff
@@ -507,17 +542,31 @@ public NextAction onSuccess(Packet packet, CallResponse<String> callResponse) {
           jobConditionsReason.add(DomainStatusUpdater.ERR_INTROSPECTOR);
         }
         //Introspector job is incomplete, update domain status and terminate processing
+        Step nextStep = null;
+        if (System.currentTimeMillis() > getJobLazyDeletionTime(domainIntrospectorJob)) {
+          //Introspector job is incomplete and current time is greater than the lazy deletion time for the job,
+          //update the domain status and execute the next step
+          nextStep = getNext();
+        }
+
         return doNext(
                 DomainStatusUpdater.createFailureRelatedSteps(
                         onSeparateLines(jobConditionsReason),
                         onSeparateLines(severeStatuses),
-                        null),
+                        nextStep),
                 packet);
       }
 
       return doNext(packet);
     }
 
+    private Long getJobLazyDeletionTime(V1Job domainIntrospectorJob) {
+      int retryIntervalSeconds = TuningParameters.getInstance().getMainTuning().domainPresenceRecheckIntervalSeconds;
+      return Optional.ofNullable(domainIntrospectorJob.getMetadata())
+              .map(m -> m.getCreationTimestamp()).map(t -> t.toInstant().toEpochMilli()).orElse(0L)
```

Comment:
Why are you converting these OffsetDateTime instances to epoch millis? Why not return an OffsetDateTime and use the isBefore() or isAfter() comparison methods. Similarly, the class has built-in support for adding seconds or millis, etc.

Reply:
Done, made the changes to use OffsetDateTime, isAfter() and plus() methods instead of converting it to epoch millis in fedcece .